src/placement/ring_map.c

```diff
@@ -1156,6 +1156,18 @@ ring_obj_remap_shards(struct pl_ring_map *rimap, struct daos_obj_md *md,
 			if (spare_tgt->ta_comp.co_fseq < f_shard->rfs_fseq)
 				continue; /* try next spare */
 
+			/* pl_obj_shard::po_fseq records the latest failure
+			 * sequence of the whole remap chain for the given
+			 * shard. On the other hand, the target (spare_tgt)
+			 * failure sequence (co_fseq) is the snapshot of the
+			 * pool map version (that is incremental only) when
+			 * the target becomes down, so the latest failure
+			 * sequence is also the larget one for the targets
+			 * on the whole remap chain.
+			 */
+			if (l_shard->po_fseq < spare_tgt->ta_comp.co_fseq)
+				l_shard->po_fseq = spare_tgt->ta_comp.co_fseq;
+
```

Comment:
Above comments and the newly added two lines of code are redundant, can be removed.

Reply:
The comment will help others to understand how the pl_obj_shard::po_fseq will be set.
If remove the new two lines, then how to set the l_shard->po_fseq? The original assignment at the line 1194 is wrong.

Reply:
The comment of 'po_fseq' already explained it's the latest fseq, above new comment looks bit confusing to me actually.
The line 1194 could be changed to 'l_shard->po_fseq = f_shard->rfs_fseq /* latest fseq of this shard */'

Reply:
no, the l_shard->po_fseq should record the latest unavailable spare's failure sequence, not the available one.

Reply:
According to our discussion, the available spare's failure sequence is always the initial value (if not consider the re-integration case).

Reply:
Isn't 'f_shard->rfs_fseq' the latest fseq?